packages/bus/src/Brokers/RMQEventBus.spec.ts

```diff
@@ -0,0 +1,560 @@
+  const mockedConnectionFactory = mock<ConnectionFactory<Connection>>();
+  const mockedConnection = mock<Connection>();
+  const mockedConfirmChannel = mock<ConfirmChannel>();
+
+  beforeEach(() => {
+    when(
+      mockedConfirmChannel.consume(anything(), anything(), anything())
+    ).thenResolve({ consumerTag: `agent:${RundomString(22)}` } as any);
+
+    when(
+      mockedConfirmChannel.assertExchange(anything(), anything(), anything())
+    ).thenResolve();
+
+    when(
+      mockedConfirmChannel.assertQueue(anything(), anything())
+    ).thenResolve();
+
+    const confirmChannel = resolvableInstance(mockedConfirmChannel);
+    when(mockedConnection.createConfirmChannel()).thenResolve(
+      confirmChannel as any
+    );
+
+    const connection = resolvableInstance(mockedConnection);
+    when(mockedConnectionFactory.create(anything(), anything())).thenResolve(
+      connection
+    );
+
+    config = new Configuration({
+      cluster: 'amqps://test-url.com:5672',
+      exchange: 'EventBus',
+      clientQueue: 'agent:nnCF9MfHpbvdJVtSbQfKa1',
+      connectTimeout: 10000,
+      heartbeatInterval: 5000,
+      reconnectTime: 5000,
+      credentials: {
+        token: 'weobbz5.nexa.vennegtzr2h7urpxgtksetz2kwppdgj0'
+      }
+    });
+  });
+
+  afterEach(() => {
+    reset(mockedConnectionFactory);
+    reset(mockedConnection);
+    reset(mockedConfirmChannel);
+  });
+
+  describe('init', () => {
+    let eventBus: EventBus;
+    beforeEach(() => {
+      eventBus = new RMQEventBus(config, instance(mockedConnectionFactory));
+    });
+
+    it('should init event bus', async () => {
+      await expect(eventBus.init?.call(eventBus)).resolves.not.toThrow();
+      verify(mockedConnectionFactory.create(anything(), anything())).once();
+      verify(mockedConnection.createConfirmChannel()).once();
+      verify(
+        mockedConfirmChannel.consume(anything(), anything(), anything())
+      ).twice();
+      verify(
+        mockedConfirmChannel.assertExchange(anything(), anything(), anything())
+      ).once();
+      verify(mockedConfirmChannel.assertQueue(anything(), anything())).once();
+    });
+
+    it('should throw if create failsed', async () => {
+      when(mockedConnectionFactory.create(anything(), anything())).thenReject();
+      await expect(eventBus.init?.call(eventBus)).rejects.toThrow();
+      verify(mockedConnectionFactory.create(anything(), anything())).atLeast(1);
+      verify(mockedConnection.createConfirmChannel()).never();
+      verify(
+        mockedConfirmChannel.consume(anything(), anything(), anything())
+      ).never();
+      verify(
+        mockedConfirmChannel.assertExchange(anything(), anything(), anything())
+      ).never();
+      verify(mockedConfirmChannel.assertQueue(anything(), anything())).never();
+    });
+  });
+
+  describe('register', () => {
+    let eventBus: EventBus;
+    beforeEach(() => {
+      eventBus = new RMQEventBus(config, instance(mockedConnectionFactory));
+    });
```

Comment:
It is still not clear why do you create this instance in each suite separately.

```suggestion
    let eventBus!: EventBus;
    beforeEach(() => {
      eventBus = new RMQEventBus(config, instance(mockedConnectionFactory));
    });
```